crates/forge-runner/src/gas.rs

```diff
@@ -1,79 +1,129 @@
+use blockifier::abi::constants;
 use std::collections::HashMap;
 
 use crate::test_case_summary::{Single, TestCaseSummary};
-use blockifier::fee::fee_utils::calculate_tx_l1_gas_usage;
-use blockifier::fee::gas_usage::calculate_tx_gas_usage;
-use blockifier::fee::os_resources::OS_RESOURCES;
-use blockifier::fee::os_usage::get_additional_os_resources;
-use blockifier::state::cached_state::{CachedState, StateChangesCount};
-use blockifier::state::errors::StateError;
-use blockifier::transaction::transaction_types::TransactionType;
-use blockifier::{
-    abi::constants, block_context::BlockContext, execution::entry_point::ExecutionResources,
-    transaction::objects::ResourcesMapping,
+use blockifier::context::TransactionContext;
+use blockifier::fee::eth_gas_constants;
+use blockifier::fee::fee_utils::calculate_tx_gas_vector;
+use blockifier::fee::gas_usage::{
+    get_consumed_message_to_l2_emissions_cost, get_da_gas_cost,
+    get_log_message_to_l1_emissions_cost,
 };
-use cairo_vm::vm::runners::cairo_runner::ExecutionResources as VmExecutionResources;
+use blockifier::state::cached_state::CachedState;
+use blockifier::state::errors::StateError;
+use blockifier::transaction::objects::{GasVector, HasRelatedFeeType, ResourcesMapping};
+use blockifier::utils::{u128_from_usize, usize_from_u128};
+use cairo_vm::vm::runners::builtin_runner::SEGMENT_ARENA_BUILTIN_NAME;
 use cheatnet::runtime_extensions::call_to_blockifier_runtime_extension::rpc::UsedResources;
 use cheatnet::state::ExtendedStateReader;
 
+// Put together from a few blockifier functions
+fn calculate_messages_costs(
```

Comment:
I probably could try returning `CallInfo`s alongside the resources 🤔

Reply:
Not possible, the function upstream takes only 1 possible l1 handler per txn, and all calculations are based on the assumption that we are calculating a 1 tx gas usage. 

This would under-estimate the cost if we used a couple of l1 handlers, and reused their methods